libraries/aws/common/src/aws_iot_doc_parser.c

```diff
@@ -229,23 +229,23 @@ bool IotJsonUtils_FindJsonValue( const char * pJsonDocument,
                 i++;
 
                 /* If the end of the document is reached, this isn't a match. */
-                if( i >= jsonDocumentLength )
+                if( i >= awsIotJsonDocumentLength )
                 {
                     return false;
                 }
 
                 /* Add the length of all characters in the JSON object or array. This
                  * includes the length of nested objects. */
-                while( pJsonDocument[ i ] != closeCharacter ||
-                       ( pJsonDocument[ i ] == closeCharacter && nestingLevel != 0 ) )
+                while( pAwsIotJsonDocument[ i ] != closeCharacter ||
```

Comment:
What if the value contains a `{` or `}` character?  e.g.

```
"key" : "{{{{{{{{{{{{",
"another-key" : "some other things"
```

Reply:
Thats a good catch. It will have issues. Will fix it